lib/transaction.js

```diff
@@ -211,30 +223,19 @@ class Transaction extends EventEmitter {
       } catch (e) {
         reject(e);
       }
-    })
-      .then((connection) => {
+    }).then(async (connection) => {
+      try {
         connection.__knexTxId = this.txid;
-
-        return (
-          this._previousSibling
-            // .catch(() => {}) // TODO: Investigate this line
-            .then(function() {
-              return connection;
-            })
-        );
-      })
-      .then(async (connection) => {
-        try {
-          return await cb(connection);
-        } finally {
-          if (!configConnection) {
-            debug('%s: releasing connection', this.txid);
-            this.client.releaseConnection(connection);
-          } else {
-            debug('%s: not releasing external connection', this.txid);
-          }
+        return await cb(connection);
+      } finally {
+        if (!configConnection) {
+          debug('%s: releasing connection', this.txid);
+          this.client.releaseConnection(connection);
+        } else {
+          debug('%s: not releasing external connection', this.txid);
         }
-      });
+      }
+    });
```

Comment:
Dang... it looks like the `git` commit hooks automatically fixed the indentation.  (I was hoping to leave the indentation as-is, and fix it in a separate PR.  Oh well.)